src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java

```diff
@@ -263,9 +259,36 @@ private BuildConfigProjectProperty buildConfigProjectForJob(WorkflowJob job) {
 
   private void upsertBuildConfigForJob(WorkflowJob job, BuildConfigProjectProperty buildConfigProjectProperty) {
     boolean create = false;
+
+    logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
+      buildConfigProjectProperty.getName());
+
     BuildConfig jobBuildConfig = getOpenShiftClient().buildConfigs().
             inNamespace(buildConfigProjectProperty.getNamespace()).withName(buildConfigProjectProperty.getName()).get();
+
+    if (jobBuildConfig == null){
+
+      logger.info("Not able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
+        buildConfigProjectProperty.getName());
+
+      logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() +
+        " with label gitRepository: " + buildConfigProjectProperty.getName());
+
```

Comment:
thanks for the explanation in the PR review @piyush1594 but I want a comment here in the code so that if I look at this a few months from now, I do not have to hunt around commits and PRs to remind myself what the heck is going on.

Reply:
I will add the comments in code too

Reply:
ok good thanks

Reply:
resolved